### PR TITLE
working ensemble slice sampling

### PR DESCRIPTION
**Pull request by wdpozzo**
I implemented the ensemble slice sampling from https://arxiv.org/pdf/2002.06212v1.pdf. It seems to be stupidly efficient and passes  all tests up to d = 20

### CODE REVIEW

**Comment by wdpozzo** [comment]
Incidentally, it turns out that logZ is unbiased if poolsize = 2*(dim + int(log2(dim))

**Comment by wdpozzo** [comment]
@johnveitch : I pushed some fixes to this branch and now it passes all tests. The 50D gaussian evidence is unbiased and runs in ~10 minutes with 12 processes. Can you please review this and approve the pull request if you are happy?

File: cpnest/cpnest.py [hunk 1]
```diff
@@ -57,17 +62,29 @@ class CPNest(object):
...
+            number of sampler threads using an ensemble slice samplers. Default: 0
+            
     resume: `boolean`
         determines whether cpnest will resume a run or run from scratch. Default: False.
 
     proposal: `dict`
         dictionary of lists with custom jump proposals.
         key 'mhs' for the Metropolis-Hastings sampler,
-        'hmc' for the Hamiltonian Monte-Carlo sampler. Default: None
+<<<<<<< HEAD
```
Comment (johnveitch) [inline cpnest/cpnest.py]: Looks like something went wrong here with a merge

File: cpnest/nest2pos.py [hunk 2]
```diff
@@ -3,6 +3,8 @@
 from numpy import logaddexp, vstack
 from numpy.random import uniform
 from functools import reduce
+from scipy.stats import pearsonr
+#from statsmodels.tsa.ar_model import AutoReg
```
Comment (johnveitch) [inline cpnest/nest2pos.py]: remove comment

File: cpnest/nest2pos.py [hunk 3]
```diff
@@ -170,15 +172,34 @@ def resample_mcmc_chain(chain, verbose=False, burnin=False):
...
+    """
+    T=1
+    i=1
+    acf = autocorrelation(x)
+    while acf[i]>tolerance and i<len(x):
+        T+=2*acf[i]
+        i+=1
+    return T
+
+#def acl_AR(x):
```
Comment (johnveitch) [inline cpnest/nest2pos.py]: remove commented code

**Review by johnveitch — CHANGES_REQUESTED** [review]
Excellent. I think you need to do a rebase or a merge to fix conflicts with master and there's a stray conflict in one of the files in the patch. I noted a couple of places  to remove commented code if it's not still useful.

**Comment by wdpozzo** [comment]
@johnveitch : I rebased against master and verified that it is still working. I also took care of your comments.

File: cpnest/nest2pos.py (hunk 3 above)
Comment (wdpozzo) [reply cpnest/nest2pos.py]: done

File: cpnest/nest2pos.py (hunk 2 above)
Comment (wdpozzo) [reply cpnest/nest2pos.py]: done

File: cpnest/cpnest.py (hunk 1 above)
Comment (wdpozzo) [reply cpnest/cpnest.py]: fixed

**Comment by johnveitch** [comment]
I added a couple of test programs for the slice sampler and all looks good!

**Pull request merged**